### PR TITLE
Database Setup

### PR DESCRIPTION
**Pull request by emilymendelson**
This is related to the following tasks:

- https://deloittedigital.atlassian.net/browse/BVPS-126
- https://deloittedigital.atlassian.net/browse/BVPS-127

Files Changed:

- local-db/README.md - Instructions on how to run database SQL locally
- local-db/init-pg.sql - Contains SQL scripts for creating user table and active_pin table
- local-db/Dockerfile
- local-db/docker-compose.yml - to build local database

The create table SQL commands will be updated once LTSA data is received.

### CODE REVIEW

**Review by ceciliahangzhou — APPROVED** [review]

File: local-db/init-pg.sql [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    first_name VARCHAR(50),
+    last_name VARCHAR(75),
+    display_name VARCHAR(125),
+    CHECK (bceid_user_guid IS NOT NULL OR idir_user_guid IS NOT NULL)
+);
+
+-- ACTIVE PIN -------------------------------------------
+-- Table Definition ----------------------------------------------
+
+CREATE TYPE parcel_status_type AS ENUM ('Active', 'Inactive');
```
Comment (hannah-macdonald1) [inline local-db/init-pg.sql:24]: Should be A, I to match the source data

**Review by hannah-macdonald1 — CHANGES_REQUESTED** [review]
One minor change. Otherwise this looks good 👍

File: local-db/init-pg.sql (hunk 1 above)
Comment (emilymendelson) [reply local-db/init-pg.sql:24]: Thank you! I'll update this :)

**Review by hannah-macdonald1 — APPROVED** [review]
👍

**Review by ceciliahangzhou — APPROVED** [review]

File: local-db/init-pg.sql [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+-- ACTIVE PIN -------------------------------------------
+-- Table Definition ----------------------------------------------
+
+CREATE TYPE parcel_status_type AS ENUM ('A', 'I');
+CREATE TYPE title_status_type AS ENUM ('R', 'C');
+
+CREATE TABLE IF NOT EXISTS "active_pin" (
+    live_pin_id UUID PRIMARY KEY NOT NULL,
+    pin VARCHAR(8) NOT NULL,
```
Comment (hannah-macdonald1) [inline local-db/init-pg.sql:29]: Based on Brett's updated requirements (recording a homeowner but not immediately issuing a PIN) this cannot be a required (NOT NULL) field

File: local-db/init-pg.sql [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+);
+
+
+-- PERMISSION -------------------------------------------
+-- Table Definition ----------------------------------------------
+
+CREATE TABLE IF NOT EXISTS "permission" (
+    permission_id UUID PRIMARY KEY NOT NULL,
+    permission VARCHAR(50) NOT NULL,
+    role VARCHAR(12) NOT NULL
```
Comment (hannah-macdonald1) [inline local-db/init-pg.sql:59]: This should probably be an ENUM type instead ('Standard','Admin','SuperAdmin')

File: local-db/init-pg.sql [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+);
+
+-- LOG -------------------------------------------
+-- Table Definition ----------------------------------------------
+
+CREATE TYPE expiration_reason_type AS ENUM ('OP', 'CC', 'OR', 'CO', 'OT');
```
Comment (hannah-macdonald1) [inline local-db/init-pg.sql:65]: Maybe add a comment below the type as to what the reason codes mean so we don't forget? :)

File: local-db/init-pg.sql [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    key DATE NOT NULL, 
+    expired_at DATE, 
+    expiration_reason expiration_reason_type, 
+    sent_to_email VARCHAR(100), 
+    sent_to_phone VARCHAR(12)
+);
+
+-- TOKEN -------------------------------------------
+-- Table Definition ----------------------------------------------
+
```
Comment (hannah-macdonald1) [inline local-db/init-pg.sql:86]: This one is very likely to change once we know more about the keycloak api output, but better to have it here than not.

**Review by hannah-macdonald1 — CHANGES_REQUESTED** [review]
A few minor comments, but looks good overall.

**Comment by emilymendelson** [comment]
> A few minor comments, but looks good overall.

Thank you for going through @hannah-macdonald1! Updated :)

**Review by hannah-macdonald1 — APPROVED** [review]
👍

File: local-db/init-pg.sql [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    first_name VARCHAR(50),
+    last_name VARCHAR(75),
+    display_name VARCHAR(125),
+    CHECK (bceid_user_guid IS NOT NULL OR idir_user_guid IS NOT NULL)
+);
+
+-- ACTIVE PIN -------------------------------------------
+-- Table Definition ----------------------------------------------
+
+CREATE TYPE parcel_status_type AS ENUM ('A', 'I');
```
Comment (emilymendelson) [inline local-db/init-pg.sql:24]: Add definitions for types

File: local-db/init-pg.sql [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+-- LOG -------------------------------------------
+-- Table Definition ----------------------------------------------
+
+-- PIN EXPIRATION REASONS: 
+    -- OP - Opt-out
+    -- CC - call center pin reset (i.e., forgotten PIN)
+    -- OR- online pin reset
+    -- CO - change of ownership (title cancelled or parcel inactive)
+    -- OT - other
```
Comment (emilymendelson) [inline local-db/init-pg.sql:70]: confirm if other is still required

File: local-db/init-pg.sql [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+CREATE TYPE expiration_reason_type AS ENUM ('OP', 'CC', 'OR', 'CO', 'OT');
+
+CREATE TABLE IF NOT EXISTS "log" (
```
Comment (emilymendelson) [inline local-db/init-pg.sql:74]: Change table name to be more descriptive (i.e., pin_audit_log)

File: local-db/init-pg.sql [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+    expiration_reason expiration_reason_type, 
+    sent_to_email VARCHAR(100), 
+    sent_to_phone VARCHAR(12)
+);
+
+-- TOKEN -------------------------------------------
+-- Table Definition ----------------------------------------------
+
+CREATE TABLE IF NOT EXISTS "token" (
+    token_id UUID PRIMARY KEY NOT NULL,
```
Comment (emilymendelson) [inline local-db/init-pg.sql:95]: Drop this table for now

File: local-db/init-pg.sql [hunk 10]
```diff
@@ -0,0 +1,97 @@
+\connect postgres
+
+-- USER -------------------------------------------
+-- Table Definition ----------------------------------------------
+
+CREATE EXTENSION citext;
+CREATE TYPE role_type AS ENUM ('Standard', 'Admin', 'SuperAdmin');
+
+CREATE TABLE IF NOT EXISTS "user" (
```
Comment (emilymendelson) [inline local-db/init-pg.sql:9]: @ceciliahangzhou go through new requirements and update user table

File: local-db/init-pg.sql (hunk 8 above)
Comment (emilymendelson) [reply local-db/init-pg.sql:74]: done

File: local-db/init-pg.sql [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+    log_id UUID PRIMARY KEY NOT NULL,
+    pin VARCHAR(8) NOT NULL,
+    pid INT NOT NULL,
+    parcel_status parcel_status_type NOT NULL,
+    title_number VARCHAR(11) NOT NULL, 
+    land_title_district VARCHAR(2) NOT NULL,
+    from_title_number VARCHAR(11),
+    from_land_title_district VARCHAR(2),
+    title_status title_status_type NOT NULL,
+    expired_at DATE,
```
[the remark this answers is not in the document]
Comment (ceciliahangzhou) [reply local-db/init-pg.sql]: Updated

**Review by hannah-macdonald1 — APPROVED** [review]
:+1:

**Review by habibaz — APPROVED** [review]

**Pull request merged**